src/github.com/matrix-org/go-neb/polling/polling.go

```diff
@@ -71,6 +72,15 @@ func pollLoop(service types.Service, ts int64) {
 		"service_id":   service.ServiceID(),
 		"service_type": service.ServiceType(),
 	})
+
+	defer func() {
+		if r := recover(); r != nil {
+			logger.WithField("panic", r).Errorf(
+				"pollLoop panicked!\n%s", debug.Stack(),
+			)
+		}
+	}()
```

Comment:
Thought about it, and no, it doesn't. I'm treating panicking code as unrecoverable, and not executing the exact same code again hoping for a different outcome.